website/docs/r/stack_hci_logical_network.html.markdown

```diff
@@ -0,0 +1,123 @@
+---
+subcategory: "Azure Stack HCI"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_stack_hci_logical_network"
+description: |-
+  Manages an Azure Stack HCI Logical Network.
+---
+
+# azurerm_stack_hci_logical_network
+
+Manages an Azure Stack HCI Logical Network.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "example-rg"
+  location = "West Europe"
+}
+
+resource "azurerm_stack_hci_logical_network" "example" {
+  name                = "example-hci-ln"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  custom_location_id  = "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg1/providers/Microsoft.ExtendedLocation/customLocations/cl1"
+  vm_switch_name      = "ConvergedSwitch(managementcompute)"
+  dns_servers         = ["10.0.0.7", "10.0.0.8"]
+
+  subnet {
+    ip_allocation_method = "Static"
+    address_prefix       = "10.0.0.0/24"
+    ip_pool {
+      start = "10.0.0.218"
+      end   = "10.0.0.230"
+    }
+    route {
+      name                = "example-route"
+      address_prefix      = "10.0.0.0/28"
+      next_hop_ip_address = "10.0.20.1"
+    }
+    vlan_id = 123
+  }
+
+  tags = {
+    foo = "bar"
+  }
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name which should be used for this Azure Stack HCI Logical Network. Changing this forces a new resource to be created.
+
+* `resource_group_name` - (Required) The name of the Resource Group where the Azure Stack HCI Logical Network should exist. Changing this forces a new resource to be created.
+
+* `location` - (Required) The Azure Region where the Azure Stack HCI Logical Network should exist. Changing this forces a new resource to be created.
+
+* `custom_location_id` - (Required) The ID of Custom Location where the Azure Stack HCI Logical Network should exist. Changing this forces a new resource to be created.
+
+* `vm_switch_name` - (Required) The name of the network switch used to associate with the Azure Stack HCI Logical Network. Possible switch name can be retrieved following the [Azure documents](https://learn.microsoft.com/azure-stack/hci/manage/create-logical-networks?tabs=azurecli#prerequisites). Changing this forces a new resource to be created.
```

Comment:
from the description, it seems should be named as `network_switch_name` ?

Reply:
changed to `virtual_switch_name` and the description to `Name of the external virtual switch`